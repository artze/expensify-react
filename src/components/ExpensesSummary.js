import React from 'react';
import { connect } from 'react-redux';
import numeral from 'numeral';
import filterExpenses from '../selectors/expenses';
import getExpensesTotal from '../selectors/expenses-total';

export const ExpensesSummary = (props) => (
    <h1>Viewing {props.expenseCount} {props.expenseCount > 1 ? 'expenses' : 'expense'} totalling {numeral(props.expenseTotal / 100).format('$0,0.00')}</h1>
)

const mapStateToProps = (state) => {
    return {
        expenseCount: filterExpenses(state.expenses, state.filters).length,
        expenseTotal: getExpensesTotal(filterExpenses(state.expenses, state.filters))
    }
}

export default connect(mapStateToProps)(ExpensesSummary);