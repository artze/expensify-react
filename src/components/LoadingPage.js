import React from 'react';
import loadingSpinner from '../assets/images/loader.gif'

const LoadingPage = () => (
    <div className="loader">
        <img className="loader__image" src={loadingSpinner} />
    </div>
)

export default LoadingPage;