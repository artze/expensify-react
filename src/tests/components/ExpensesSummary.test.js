import React from 'react';
import { shallow } from 'enzyme';
import { ExpensesSummary } from '../../components/ExpensesSummary';

test('should render 1 expense correctly', () => {
    const wrapper = shallow(<ExpensesSummary expenseCount={1} />)
    expect(wrapper).toMatchSnapshot();
})

test('should render multiple expense correctly', () => {
    const wrapper = shallow(<ExpensesSummary expenseCount={4} />)
    expect(wrapper).toMatchSnapshot();
})