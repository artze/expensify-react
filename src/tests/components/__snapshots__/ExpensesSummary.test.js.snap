// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render 1 expense correctly 1`] = `
<h1>
  Viewing 
  1
   
  expense
   totalling 
  $0.00
</h1>
`;

exports[`should render multiple expense correctly 1`] = `
<h1>
  Viewing 
  4
   
  expenses
   totalling 
  $0.00
</h1>
`;
